src/MutableMap/MutableMap.dfy

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT
+*******************************************************************************/
+
+// RUN: %dafny /compile:0 "%s"
+
+module {:extern "MutableMap"} {:options "/functionSyntax:4"} MutableMap {
+  class {:extern "MutableMap"} MutableMap<K(==),V(==)> {
+    function {:extern "content"} content(): map<K, V>
+      reads this
+
+    constructor {:extern "MutableMap"} ()
+      ensures this.content() == map[]
```

Comment:
Generally better to provide a static factory method as part of your API instead of a constructor, so the concrete type can be changed without breaking anyone.

This suggests making `MutableMap` a trait rather than a class is a good idea, which will also help with the feasibility implementation as per my comment there.

Reply:
Discussed this with @atomb, didn't work out as hoped. I suggest we leave it as it is and potentially address this again in the future.